hatchet_sdk/context/context.py

```diff
@@ -376,6 +376,9 @@ def child_key(self) -> str | None:
     def parent_workflow_run_id(self) -> str | None:
         return self.action.parent_workflow_run_id
 
+    def step_run_errors(self) -> dict[str, str]:
+        return cast(dict[str, str], self.data.get("step_run_errors", {}))
```

Comment:
If we are running against an old engine version which doesn't pass `step_run_errors`, should we throw an exception to the user? And should we figure out if this is an on failure step and throw an exception if it's not?

Reply:
ah that's a fair question, seems like a reasonable idea to me. did @reillyse just add a way to query for the engine version?

and yeah, throwing if it's not an on-failure step seems reasonable to me 👍

Reply:
I don't believe we have the engine version on the SDKs at the moment, there is a way to query for the API version but it's not guaranteed to be the same as the engine version. 

Honestly this is probably fine, there's not a clean way to determine that this is an on failure step against an old engine version, so we can't really do the casing properly anyway. 

Perhaps we just throw an exception if this doesn't exist, and we include that (1) you can't call this from non-onfailure steps, and (2) people should upgrade their SDK version to <version> and their engine version to <version>

Reply:
one possible footgun is that if someone tries to use this on an old engine version in production and it breaks, the rest of their on failure step won't work if we throw an exception, as opposed to if we `logger.error` or something (that should get bubbled up to e.g. Sentry but also not crash their job). WDYT of that?

Reply:
yeah that makes sense to me